servers/rendering/renderer_viewport.cpp

```diff
@@ -526,6 +526,67 @@ void RendererViewport::_draw_viewport(Viewport *p_viewport) {
 	}
 }
 
+bool RendererViewport::_check_viewport_visible(Viewport *p_viewport, const uint64_t p_pass, Ref<XRInterface> &p_xr_interface) {
+	if (p_viewport->last_pass == p_pass && p_pass) {
+		return p_viewport->visible; // Returns the computed cached value.
+	}
+
+	p_viewport->last_pass = p_pass;
+	p_viewport->visible = false;
+
+	if (p_viewport->update_mode == RS::VIEWPORT_UPDATE_DISABLED) {
+		return false;
+	}
+
+	if (!p_viewport->render_target.is_valid()) {
+		return false;
+	}
+	//ERR_CONTINUE(!p_viewport->render_target.is_valid());
+
+	p_viewport->visible = p_viewport->viewport_to_screen_rect != Rect2();
+
+	if (p_viewport->use_xr) {
+		if (p_xr_interface.is_valid()) {
```

Comment:
Should change this to `if (p_viewport->use_xr && p_xr_interface.is_valid()) {`
Then we can put some of the other checks inside of an `else`. With XR viewports we ignore `update_mode`.